### PR TITLE
fix #46: add example violating certificates

### LINKED ISSUE #46: sample certs that violate BRs
We should make sample certs that violate BRs available through the UI at some point, through searching by root CA.

### PR DESCRIPTION
**Pull request by mozkeeler**
For some reason I initially made this way more difficult than necessary, but I eventually arrived at a reasonable solution. For now the example certs are just PEM that you have to pipe through `openssl x509 -text -noout`, but I'm looking at some JS libraries that can parse certificates so we can expose the relevant information inline.

### CODE REVIEW

**Comment by mozkeeler** [comment]
@monicachew this is ready for review

**Comment by monicachew** [comment]
Thanks for working on this. From my reading, it looks like once we hit one example, we don't update the example map, right?

I think this will have the effect that we will prefer early examples rather than later, which might be problematic for CAs that are trying to use this for seeing how their stuff breaks over time. Is it too high overhead to do a replacement strategy instead, so that later examples have a chance of displacing earlier ones?

Speaking of earlier vs. later, is it important to keep track of when we last saw the example?

**Comment by monicachew** [comment]
This is called sampling with replacement, you might be able to use this if it's not too complicated

http://en.wikipedia.org/wiki/Reservoir_sampling

File: tools/sunlight.go [hunk 1]
```diff
@@ -263,6 +306,17 @@ func main() {
...
 				os.Exit(1)
 			}
+
+			exampleMapLock.Lock()
+			if exampleMap[cert.Issuer.CommonName] == nil {
+				exampleMap[cert.Issuer.CommonName] = make(map[string]*x509.Certificate)
+			}
+			for violation, isViolation := range summary.Violations {
+				if isViolation {
+					exampleMap[cert.Issuer.CommonName][violation] = cert
```
Comment (mozkeeler) [inline tools/sunlight.go]: Unless I'm not understanding what I wrote, this line will overwrite any previous example issued by that issuer for that violation, right?

**Comment by mozkeeler** [comment]
Yes, I think it might be useful to keep track of when each example was added to the log. I'll go ahead and add that in.

**Comment by monicachew** [comment]
Ah, well then please disregard my comment about replacement sampling. I misread.

**Comment by mozkeeler** [comment]
Oh, also I put this up at https://people.mozilla.org/~dkeeler/dashboard/ so you can see how it looks.

**Comment by monicachew** [comment]
Thanks! Looks great for a first cut.

**Comment by mozkeeler** [comment]
I added the last seen timestamps if you want to have a quick look. Thanks! (I also deployed it to the same url if you want to see how it looks)

**Comment by monicachew** [comment]
Great, thanks!

**Pull request merged**